src/SME.SGP.Aplicacao/Consultas/ConsultasEventoTipo.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.AspNetCore.Http;
+using SME.SGP.Aplicacao.Interfaces.Consultas;
+using SME.SGP.Dominio.Entidades;
+using SME.SGP.Dominio.Interfaces.Repositorios;
+using SME.SGP.Dto;
+using SME.SGP.Infra;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SME.SGP.Aplicacao.Consultas
+{
+    public class ConsultasEventoTipo : ConsultasBase, IConsultasEventoTipo
+    {
+        private readonly IRepositorioEventoTipo repositorioEventoTipo;
+
+        public ConsultasEventoTipo(IRepositorioEventoTipo repositorioEventoTipo, IHttpContextAccessor httpContext) : base(httpContext)
+        {
+            this.repositorioEventoTipo = repositorioEventoTipo ?? throw new ArgumentNullException(nameof(repositorioEventoTipo));
+        }
+
+        public async Task<PaginacaoResultadoDto<EventoTipoDto>> Listar(FiltroEventoTipoDto Filtro)
+        {
+            var retornoQueryPaginada = await repositorioEventoTipo.ListarTipos(Filtro.LocalOcorrencia, Filtro.Letivo, Filtro.Descricao, Paginacao);
+
+            var retornoConsultaPaginada = new PaginacaoResultadoDto<EventoTipoDto>
+            {
+                TotalPaginas = retornoQueryPaginada.TotalPaginas,
+                TotalRegistros = retornoQueryPaginada.TotalRegistros
+            };
```

Comment:
validar retornoQueryPaginada == null quando for definir TotalPaginas e TotalRegistros